api/coverage.py

```diff
@@ -70,9 +70,32 @@ def process_batch(self, batch):
         )
 
         results = []
+        leftover_identifiers = set()
+        # We grant coverage if an Edition was created from the operation.
         for edition in imported_editions:
-            self.finalize_edition(edition)
-            results.append(edition.primary_identifier)
+            identifier = edition.primary_identifier
+            results.append(identifier)
+            leftover_identifiers.add(identifier)
+
+        # We may also have created a LicensePool from the operation.
+        for pool in pools:
+            self.finalize_license_pool(pool)
+            identifier = pool.identifier
+            if identifier in leftover_identifiers:
+                leftover_identifiers.remove(identifier)
+            else:
+                msg = "OPDS import operation imported LicensePool, but no Edition."
+                results.append(
+                    CoverageFailure(
+                        identifier, msg, data_source=self.output_source,
+                        transient=True
+                    )
+                )
+        for identifier in leftover_identifiers:
```

Comment:
This is a little weird but not weird enough to be an error. It just means the data source has metadata about the book but no copies of the book.